SD_Assignment1/src/main/java/ro/utcn/aron/service/QuestionManagementService.java

```diff
@@ -0,0 +1,38 @@
+package ro.utcn.aron.service;
+
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import ro.utcn.aron.api.QuestionRepository;
+import ro.utcn.aron.api.RepositoryFactory;
+import ro.utcn.aron.model.Question;
+
+@Component
+public class QuestionManagementService {
+
+	private  RepositoryFactory repositoryFactory; //final
+		
+	@Transactional
+	public List<Question> listQuestions(){
+		QuestionRepository repository = repositoryFactory.createQuestionRepository();
+		return repository.findAll();
+	}
+	
+	@Transactional
+	public void removeQuestion(int id) {
+		QuestionRepository repository = repositoryFactory.createQuestionRepository();
+		Question question = repository.findByID(id).orElseThrow(RuntimeException::new);

```

Comment:
[Optional] It is usually better to declare your own exceptions (subclass of runtime exception). For example "QuestionNotFoundException".